app/src/main/java/com/github/steroidteam/todolist/view/ItemViewFragment.java

```diff
@@ -305,6 +321,77 @@ public void checkBoxTaskListener(final int position, final boolean isChecked) {
         viewModel.setTaskDone(position, isChecked);
     }
 
+    public void tagButton(View view) {
+        tags = viewModel.getTags();
+        tags.sort(Tag.sortByBody);
+        ConstraintLayout tagLayout = getView().findViewById(R.id.layout_update_tags);
+        tagLayout.setVisibility(View.VISIBLE);
+        LinearLayout row = getView().findViewById(R.id.tag_row_first);
+        row.removeAllViews();
+        Button plusButton =
+                new Button(new ContextThemeWrapper(getContext(), R.style.TagInList), null, 0);
+        plusButton.setText("+");
+        plusButton.setBackgroundTintList(ColorStateList.valueOf(Color.LTGRAY));
+        plusButton.setOnClickListener(this::createTag);
+        row.addView(plusButton);
+        tags.forEach(tag -> createTagButton(tag, row));
+    }
+
+    public void tagSaveButton(View view) {
+        ConstraintLayout tagLayout = getView().findViewById(R.id.layout_update_tags);
+        tagLayout.setVisibility(View.GONE);
+    }
+
+    private void createTagButton(Tag tag, LinearLayout row) {
+        Button tagButton =
+                new Button(new ContextThemeWrapper(getContext(), R.style.TagInList), null, 0);
+        tagButton.setText(tag.getBody());
+        tagButton.setBackgroundTintList(ColorStateList.valueOf(tag.getColor()));
+        // tagButton.setBackgroundColor(;
+        tagButton.setOnLongClickListener(
+                view -> {
+                    destroyTag(tag);
+                    row.removeView(tagButton);
+                    return true;
+                });
+        row.addView(tagButton);
+    }
+
+    public void createTag(View view) {
+
+        DialogListener dialogListener =
+                new DialogListener() {
+
+                    @Override
+                    public void onPositiveClick(String title) {
+                        if (title.length() > 0) {
+                            Tag tag = new Tag(title);
+                            viewModel.addTag(tag);
+                            createTagButton(tag, getView().findViewById(R.id.tag_row_first));
+                        }
+                    }
+
+                    @Override
+                    public void onPositiveClick() {
+                        // NEVER CALLED
+                    }
+
+                    @Override
+                    public void onNegativeClick() {
+                        // DO NOTHING
+                    }
+                };
+
+        DialogFragment newFragment =
+                new InputDialogFragment().newInstance(dialogListener, R.string.add_tag_suggestion);
+        newFragment.show(getParentFragmentManager(), "add_tag_dialog");
+    }
+
+    public void destroyTag(Tag tag) {
+        // TODO : Uncomment after fixing method in repository/database
+        // itemViewModel.destroyTag(tag);
+    }
+
```

Comment:
I tested and you can't delete a tag, I think it might come from this commented code